### PR TITLE
Fix scale down condition to exclude skipped

### LINKED ISSUE #1326: The webhook based autoscaler seems scaled down for skipped job.
**Describe the bug**

When GitHub post a `workflow_job` event with conclusion `skipped`, the scaler scaled down my HRA by -1, which is a unexpected scaling, as the skipped event is not paired with a `queue` event.

I guess this is the searching logic for scale target is not handling the case where the labels are `[]`.  Please see the examples below, we can find that for these four conclusion `skipped`, `canceled`, `success`, and `failure`, only the `labels` field in the `skipped` payload is empty, leading to an incorrect scale down.
It seems to me that if the `labels` of the event are empty, it will always return the scale target to the caller.

https://github.com/actions-runner-controller/actions-runner-controller/blob/f907f822754b830d45791b52ca0f93f56d773b8f/controllers/horizontal_runner_autoscaler_webhook.go#L724-L746

This is the payload for `skipped`.
```json
{
  "action": "completed",
  "workflow_job": {
    "id": 5850345297,
    "run_id": 2102178157,
    "run_url": "https://api.github.com/repos/siyuan0322/GraphScope/actions/runs/2102178157",
    "run_attempt": 1,
    "node_id": "CR_kwDOEyo4jc8AAAABXLUvUQ",
    "head_sha": "cc723d28f438a8b8e9c6b2fd81ca50af870baab3",
    "url": "https://api.github.com/repos/siyuan0322/GraphScope/actions/jobs/5850345297",
    "html_url": "https://github.com/siyuan0322/GraphScope/runs/5850345297?check_suite_focus=true",
    "status": "completed",
    "conclusion": "skipped",
    "started_at": "2022-04-06T11:59:48Z",
    "completed_at": "2022-04-06T11:59:48Z",
    "name": "helm-test",
    "steps": [

    ],
    "check_run_url": "https://api.github.com/repos/siyuan0322/GraphScope/check-runs/5850345297",
    "labels": [

    ],
    "runner_id": null,
    "runner_name": null,
    "runner_group_id": null,
    "runner_group_name": null
  }
```

This is the payload for `cancelled`.
```json
  "workflow_job": {
    "id": 5850104853,
    "run_id": 2102178160,
    "run_url": "https://api.github.com/repos/siyuan0322/GraphScope/actions/runs/2102178160",
    "run_attempt": 1,
    "node_id": "CR_kwDOEyo4jc8AAAABXLGEFQ",
    "head_sha": "cc723d28f438a8b8e9c6b2fd81ca50af870baab3",
    "url": "https://api.github.com/repos/siyuan0322/GraphScope/actions/jobs/5850104853",
    "html_url": "https://github.com/siyuan0322/GraphScope/runs/5850104853?check_suite_focus=true",
    "status": "completed",
    "conclusion": "cancelled",
    "started_at": "2022-04-06T12:00:18Z",
    "completed_at": "2022-04-06T12:00:18Z",
    "name": "build-wheels",
    "steps": [

    ],
    "check_run_url": "https://api.github.com/repos/siyuan0322/GraphScope/check-runs/5850104853",
    "labels": [
      "self-hosted",
      "manylinux2014"
    ],
    "runner_id": null,
    "runner_name": null,
    "runner_group_id": null,
    "runner_group_name": null
  }
```

This is for `failure`.
```json
  "workflow_job": {
    "id": 5844083920,
    "run_id": 2099794750,
    "run_url": "https://api.github.com/repos/siyuan0322/GraphScope/actions/runs/2099794750",
    "run_attempt": 1,
    "node_id": "CR_kwDOEyo4jc8AAAABXFWk0A",
    "head_sha": "2a654ec05284ae63d4d075ce8be2917710b923a1",
    "url": "https://api.github.com/repos/siyuan0322/GraphScope/actions/jobs/5844083920",
    "html_url": "https://github.com/siyuan0322/GraphScope/runs/5844083920?check_suite_focus=true",
    "status": "completed",
    "conclusion": "failure",
    "started_at": "2022-04-06T02:25:16Z",
    "completed_at": "2022-04-06T02:32:18Z",
    "name": "gie-test",
    "steps": [
      {
        "name": "Set up job",
        "status": "completed",
        "conclusion": "success",
        "number": 1,
        "started_at": "2022-04-06T02:25:15.000Z",
        "completed_at": "2022-04-06T02:25:21.000Z"
      },
      {
        "name": "Complete job",
        "status": "completed",
        "conclusion": "success",
        "number": 15,
        "started_at": "2022-04-06T02:32:12.000Z",
        "completed_at": "2022-04-06T02:32:12.000Z"
      }
    ],
    "check_run_url": "https://api.github.com/repos/siyuan0322/GraphScope/check-runs/5844083920",
    "labels": [
      "self-hosted",
      "ubuntu2004"
    ],
    "runner_id": 136,
    "runner_name": "graphscope-runner-ubuntu-24wl5-5ptrr",
    "runner_group_id": 1,
    "runner_group_name": "Default"
  }
```

This is for `success`.

```json
  "workflow_job": {
    "id": 5844880021,
    "run_id": 2100210617,
    "run_url": "https://api.github.com/repos/siyuan0322/GraphScope/actions/runs/2100210617",
    "run_attempt": 1,
    "node_id": "CR_kwDOEyo4jc8AAAABXGHKlQ",
    "head_sha": "548a8904e1481b796202a8f0a976117062a133ad",
    "url": "https://api.github.com/repos/siyuan0322/GraphScope/actions/jobs/5844880021",
    "html_url": "https://github.com/siyuan0322/GraphScope/runs/5844880021?check_suite_focus=true",
    "status": "completed",
    "conclusion": "success",
    "started_at": "2022-04-06T04:08:59Z",
    "completed_at": "2022-04-06T04:37:44Z",
    "name": "build-wheels",
    "steps": [
      {
        "name": "Set up job",
        "status": "completed",
        "conclusion": "success",
        "number": 1,
        "started_at": "2022-04-06T04:08:57.000Z",
        "completed_at": "2022-04-06T04:09:01.000Z"
      },
      {
        "name": "Complete job",
        "status": "completed",
        "conclusion": "success",
        "number": 15,
        "started_at": "2022-04-06T04:37:37.000Z",
        "completed_at": "2022-04-06T04:37:37.000Z"
      }
    ],
    "check_run_url": "https://api.github.com/repos/siyuan0322/GraphScope/check-runs/5844880021",
    "labels": [
      "self-hosted",
      "manylinux2014"
    ],
    "runner_id": 143,
    "runner_name": "graphscope-runners-manylinux2014-f2md9-grh9w",
    "runner_group_id": 1,
    "runner_group_name": "Default"
  }
```

**Checks**

- [x] My actions-runner-controller version (v0.x.y) does support the feature
- [x] I'm using an unreleased version of the controller I built from HEAD of the default branch

**Expected behavior**
The `skipped` event should not scale down the HRA.

**Screenshots**

**Environment (please complete the following information):**
 - Controller Version [e.g. 0.22.3]
 - Deployment Method [e.g. Helm ]
 - Helm Chart Version [e.g. 0.17.3, if applicable]

### PR DESCRIPTION
**Pull request by siyuan0322**
A webhook event which conclusion is `skipped` should not trigger a scale down operation.

Fixes #1326

### CODE REVIEW

File: controllers/horizontal_runner_autoscaler_webhook.go [hunk 1]
```diff
@@ -246,7 +246,7 @@ func (autoscaler *HorizontalRunnerAutoscalerGitHubWebhook) Handle(w http.Respons
 			if target != nil {
 				if e.GetAction() == "queued" {
 					target.Amount = 1
-				} else if e.GetAction() == "completed" {
+				} else if e.GetAction() == "completed" && e.GetConclusion() != "skipped" {
```
Comment (mumoshu) [inline controllers/horizontal_runner_autoscaler_webhook.go:249]:
```suggestion
				} else if e.GetAction() == "completed" && e.GetWorkflowJob().GetConclusion() != "skipped" {
```

Apparently we needed to drill it down to actually get conclusion

**Comment by siyuan0322** [comment]
Wait a second, seems set the amount to zero will cause another issue... 
https://github.com/actions-runner-controller/actions-runner-controller/blob/960a704246a42d7db84a469fd49e2d21733bd733/controllers/horizontal_runner_autoscaler_webhook.go#L762-L766

**Comment by mumoshu** [comment]
> Wait a second, seems set the amount to zero will cause another issue...

Ah, good catch. If it turned out to be a real issue, perhaps we may need to skip tryScale when the conclusion is `skipped` 🤔

**Comment by siyuan0322** [comment]
>  If it turned out to be a real issue, perhaps we may need to skip tryScale when the conclusion is skipped

A fallthrough seems can do this?
```go
if e.GetAction() == "queued" {
    target.Amount = 1
} else if e.GetAction() == "completed" && e.GetWorkflowJob().GetConclusion() != "skipped" {
    target.Amount = -1
} else {
    fallthrough
}
```

**Comment by mumoshu** [comment]
> A fallthrough seems can do this?

Neat trick! I've never tried using fallthrough that way(within a nested if-else) but that might work.

**Comment by siyuan0322** [comment]
I'm not very familiar with go, this syntax won't compile... I'll do more experiment and make sure it works as expected.

**Review by mumoshu — APPROVED** [review]
Good job! Thanks a lot for your contribution @siyuan0322 ☺️

**Pull request merged**